pinot-core/src/test/java/org/apache/pinot/core/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -198,6 +207,51 @@ public void testEnableTextIndexOnNewColumnRaw()
     checkTextIndexCreation(NEWLY_ADDED_STRING_COL_RAW, 1, 1, _newColumnsSchemaWithText, true, true, true, 4);
   }
 
+  @Test
```

Comment:
I think you should add another test for enabling FST on raw column (both new and existing) and verify SegmentPreprocessor fails.

Also, add tests in LoaderTest similar to text index tests.

Reply:
done, let me know if i missed something